File: Linked_Lists/addTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        smartie = ListNode()
        tail = smartie
        overFlow = 0

        while (l1 or l2 or overFlow):
            val1 = l1.val if l1 else 0
            val2 = l2.val if l2 else 0

            val = (val1 + val2 + overFlow) % 10
            overFlow = (val1 + val2 + overFlow) // 10

            tail.next = ListNode(val)

            tail = tail.next
            if l1:
                l1 = l1.next
            if l2:
                l2 = l2.next

        return smartie.next

File: Linked_Lists/addTwoNumbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        
        ListNode smartie = new ListNode();
        ListNode tail = smartie;
        int overFlow = 0, val1 = 0, val2 = 0, val = 0;

        while (l1 != null || l2 != null || overFlow != 0) {
            if (l1 != null) {
                val1 = l1.val;
            } else val1 = 0;
            if (l2 != null) {
                val2 = l2.val;
            } else val2 = 0;

            val = (val1 + val2 + overFlow) % 10;
            overFlow = (val1 + val2 + overFlow) / 10;

            tail.next = new ListNode(val);

            tail = tail.next;
            if (l1 != null) {
                l1 = l1.next;
            } else l1 = null;

            if (l2 != null) {
                l2 = l2.next;
            } else l2 = null;
        }
        return smartie.next;
    }
}